### PR TITLE
upgraded dompurify to 2.2.2

### PR DESCRIPTION
**Pull request by knguyen142**
## Summary

https://dimagi-dev.atlassian.net/browse/SAAS-11519
Upgrades DOMPurify from 2.0.12 to 2.2.2.

## Product Description

Upgrades package.

## Safety Assurance

- [x ] Risk label is set correctly
- [x ] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x ] QA labels are set correctly
- [x ] I am certain that this PR will not introduce a regression for the reasons below

This is currently on staging to be deployed to prod sometime next week. Looks like we use this extensively in [webapps](https://github.com/dimagi/commcare-hq/blob/fd3e7ac68c78e08851f7773c2ff5adae2b0e1c61/corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js#L154) so if webapps is working, I think we should be good? Someone who has more knowledge of the library might want to weigh in.

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-2105
Smoke Tests with QA, request form submitted and will update with ticket.

### Safety story

Upgrade to a newer version.

### Rollback instructions

- [x ] This PR can be reverted after deploy with no further considerations 
Frontend code.

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

**Comment by dannyroberts** [comment]
If QA is part of the plan for https://github.com/dimagi/commcare-hq/pull/28781 (and I think it should be) then I think this needs the Awaiting QA label. That was my intention in adding a "QA labels are set correctly" box, to make people ask whether they need to add Awaiting QA—perhaps I should reword it to better prompt that thought process? Or did you have that thought process and we are interpreting the label meanings differently or something? Asking both for this PR and so that we can improve the template / docs

**Comment by biyeun** [comment]
@dannyroberts i think it might be good to reword so that QA is explicitly part of the thought process in setting labels

**Review by biyeun — APPROVED** [review]

**Comment by knguyen142** [comment]
@dannyroberts I had originally thought the label only applied to the risk level, so having a comment for risk level and QA status would be good. The only reason I added it was I saw I was getting PR approvals so wanted to put in the QA label to denote a do not merge yet.

**Comment by knguyen142** [comment]
[QA done](https://dimagi-dev.atlassian.net/browse/QA-2105), unless there are any objections, I can merge and deploy later today.
